javascript/src/utils/stream.utils.ts

```diff
@@ -80,10 +85,35 @@ export class ContentDeltaAccumulator {
               (existingDelta.part.transcript || "") +
               incomingDelta.part.transcript;
           }
+        } else if (
+          existingDelta.part.type === "reasoning" &&
+          incomingDelta.part.type === "reasoning"
+        ) {
+          if (incomingDelta.part.reasoning) {
+            existingDelta.part.reasoning =
+              (existingDelta.part.reasoning || "") +
+              incomingDelta.part.reasoning;
+          }
+          if (incomingDelta.part.summary) {
+            existingDelta.part.summary = incomingDelta.part.summary;
+          }
         } else {
-          throw new Error(
-            `unexpected part at index ${String(incomingDelta.index)}. existing part has type ${existingDelta.part.type}, incoming part has type ${incomingDelta.part.type}`,
-          );
+          // For different content types at the same index (e.g., reasoning -> text in OpenAI Responses API),
+          // treat as separate parts rather than trying to merge them
+          this.deltas.push({
+            index: incomingDelta.index,
+            part: {
+              ...(incomingDelta.part.type === "audio"
+                ? {
+                    ...incomingDelta.part,
+                    audioData: incomingDelta.part.audioData
+                      ? [base64ToArrayBuffer(incomingDelta.part.audioData)]
+                      : [],
+                  }
+                : incomingDelta.part),
```

Comment:
imo, i'd prefer to avoid nesting ternary operators

in those cases, for readability, could we move this to another function/method?